daemon/daemon_unix_test.go

```diff
@@ -376,3 +379,75 @@ func sysInfo(t *testing.T, opts ...func(*sysinfo.SysInfo)) sysinfo.SysInfo {
 	}
 	return si
 }
+
+func TestGetBlkioWeightDevices(t *testing.T) {
+	tempDir, err := ioutil.TempDir("", "tempDevDirForBlkioThrottleDevices")
+	assert.NilError(t, err, "create temp file")
+
+	tempFile := filepath.Join(tempDir, "dev")
+
+	defer os.RemoveAll(tempDir)
+
+	// mock a file with major 0x1FD(509 in decimal) and minor 0x130(304)
+	if err = unix.Mknod(tempFile, unix.S_IFCHR, 0x11FD30); err != nil {
+		t.Fatalf("mknode error %s(%x): %v", tempFile, 0x11FD30, err)
+	}
+
+	mockResource := containertypes.Resources{
+		BlkioWeightDevice: []*blkiodev.WeightDevice{{Path: tempFile, Weight: 1024}},
+	}
+
+	weightDevs, err := getBlkioWeightDevices(mockResource)
+	assert.NilError(t, err, "getBlkioWeightDevices")
+
+	if len(weightDevs) != 1 {
+		t.Fatalf("getBlkioWeightDevices return slice error, length expect 1 got %d", len(weightDevs))
+	}
+
+	if weightDevs[0].Major != 509 {
+		t.Fatalf("get major device type error. got %x expect %x", weightDevs[0].Major, 509)
+	}
+
+	if weightDevs[0].Minor != 304 {
+		t.Fatalf("get minor device type error. got %x expect %x", weightDevs[0].Minor, 304)
+	}
+
+	if *weightDevs[0].Weight != 1024 {
+		t.Fatalf("get device weight error. got %x expect %x", *weightDevs[0].Weight, 1024)
+	}
+}
+
+func TestGetBlkioThrottleDevices(t *testing.T) {
```

Comment:
This second test is the same as the first one, except for function called. Can you please combine the code so it won't repeat twice?